pkg/apis/clusterlink.net/v1alpha1/operator_types.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StatusType represents the status options for ClusterLink components.
+type StatusType string
+
+const (
+	// StatusNone means a resource was not created.
+	StatusNone StatusType = "none"
+	// StatusPending means a resource was created but is not ready to use.
+	StatusPending StatusType = "pending"
+	// StatusError means an error occurred when the resource was created.
+	StatusError StatusType = "error"
+	// StatusReady means a resource was created and is ready to use.
+	StatusReady StatusType = "ready"
+)
+
+// IngressType represents the ingress type  of the deployed ClusterLink.
+type IngressType string
+
+const (
+	// IngressTypeNone indicates that the deployment operator did not create any external ingress.
+	IngressTypeNone IngressType = "none"
+	// IngressTypeNodePort indicates that the deployment operator created an external ingress of type NodePort.
+	IngressTypeNodePort IngressType = "NodePort"
+	// IngressTypeLoadBalancer indicates that the deployment operator created an external ingress of type LoadBalancer.
+	IngressTypeLoadBalancer IngressType = "LoadBalancer"
+)
+
+// DataplaneType represents the dataplane type  of the deployed ClusterLink.
+type DataplaneType string
+
+const (
+	// GoType indicates that the dataplane type is Go-dataplane.
+	GoType DataplaneType = "go"
+	// EnvoyType indicates that the dataplane type is Envoy.
+	EnvoyType DataplaneType = "envoy"
+)
+
+// OperatorSpec defines the desired state of a ClusterLink Operator.
+type OperatorSpec struct {
+	DataPlane DataPlaneSpec `json:"dataplane,omitempty"`
+	Ingress   IngressSpec   `json:"ingress,omitempty"`
+
+	// +kubebuilder:validation:Enum=trace;debug;info;warning;error;fatal
+	// +kubebuilder:default=info
+	// LogLevel define the ClusterLink components log level.
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// +kubebuilder:default="ghcr.io/clusterlink-net"
+	// ContainerRegistry is the container registry to pull the ClusterLink project images.
+	ContainerRegistry string `json:"containerRegistry,omitempty"`
+	// +kubebuilder:default="latest"
+	// ImageTag represents the version of the ClusterLink project images.
+	ImageTag string `json:"imageTag,omitempty"`
```

Comment:
will we update a running deployment of the tag is changed on an existing spec? I think users expect that (redeploy) behavior.
If not, document that this is a one time deployment and users should delete/recreate on every change. 
IIRC, there is a kubebuilder [plugin](https://book.kubebuilder.io/plugins/deploy-image-plugin-v1-alpha) to help with it. Please check if useful/appropriate for our use case.